src/Avalonia.X11/Dbus/LinuxDriveInfoProvider.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+using Avalonia.Controls.Platform;
+using Tmds.DBus;
+
+namespace Avalonia.X11.Dbus
+{
+    public class LinuxMountedDriveInfoProvider : IMountedDriveInfoProvider
+    {
+        private IDisposable[] Disposables;
+        private readonly Connection _sysDbus;
+        private readonly IObjectManager udisk2Manager;
+
+        public LinuxMountedDriveInfoProvider()
+        {
+            this._sysDbus = Connection.System;
+            this.udisk2Manager = _sysDbus.CreateProxy<IObjectManager>("org.freedesktop.UDisks2", "/org/freedesktop/UDisks2");
+        }
+
+        public ObservableCollection<MountedDriveInfo> CurrentDrives { get; } = new ObservableCollection<MountedDriveInfo>();
+
+        private bool _isMonitoring;
+        public bool IsMonitoring => _isMonitoring;
+
+        private async void Poll()
+        {
+            var newDriveList = new List<MountedDriveInfo>();
+
+            var fProcMounts = File.ReadAllLines("/proc/mounts");
+
+            var managedObj = await udisk2Manager.GetManagedObjectsAsync();
+
+            var res_drives = managedObj.Where(x => x.Key.ToString().Contains("/org/freedesktop/UDisks2/drives/"))
+                                 .Select(x => x.Key);
+
+            var res_blockdev = managedObj.Where(x => x.Key.ToString().Contains("/org/freedesktop/UDisks2/block_devices/"))
+                                   .Select(x => x);
+
+            var res_fs = managedObj.Where(x => x.Key.ToString().Contains("system"))
+                             .Select(x => x.Key)
+                             .ToList();
+
+            foreach (var block in res_blockdev)
+            {
+
+                var iblock = _sysDbus.CreateProxy<IBlock>("org.freedesktop.UDisks2", block.Key);
+                var iblockProps = await iblock.GetAllAsync();
+
+                var block_drive = await iblock.GetDriveAsync();
+                if (!res_drives.Contains(block_drive)) continue;
+
+                var drive_key = res_drives.Single(x => x == block_drive);
+                var drives = _sysDbus.CreateProxy<IDrive>("org.freedesktop.UDisks2", drive_key);
+                var drivesProps = await drives.GetAllAsync();
+
+                var devRawBytes = iblockProps.Device.Take(iblockProps.Device.Length - 1).ToArray();
+                var devPath = System.Text.Encoding.UTF8.GetString(devRawBytes);
+
+                var blockLabel = iblockProps.IdLabel;
+                var blockSize = iblockProps.Size;
+                var driveName = drivesProps.Id;
+
+                // There should be something in udisks2 to 
+                // get this data but I have no idea where.
+                var mountPoint = fProcMounts.Select(x => x.Split(' '))
```

Comment:
@kekekeks I never figured out how to use Filesystems on UDisks2 so i hope this is a good enough compromise. Do tell if you have better suggestions on this one